routing/matcher.go

```diff
@@ -487,9 +487,8 @@ func matchLeaves(leaves leafMatchers, req *http.Request, path, exactPath string)
 // returns the associated value, and the wildcard parameters from the path definition,
 // if any.
 func (m *matcher) match(r *http.Request) (*Route, map[string]string) {
-	// normalize path before matching
 	// in case ignoring trailing slashes, match without the trailing slash
-	path := httppath.Clean(r.URL.Path)
```

Comment:
There are quite some other places where `httppath.Clean()` is used in the matcher. Should they be considered as well?

Reply:
I think the other ones are cleaning a path derived from the *route which is config and not something the change in the proxy.go affects. (maybe we should clean earlier in the route definition too but thats a bit out of scope for this issue)